sormas-ui/src/main/java/de/symeda/sormas/ui/location/LocationEditForm.java

```diff
@@ -202,9 +205,26 @@ protected void addFields() {
 		initializeAccessAndAllowedAccesses();
 
 		if (!isEditableAllowed(LocationDto.COMMUNITY)) {
-			setEnabled(false, LocationDto.REGION, LocationDto.DISTRICT);
+			setEnabled(false, LocationDto.COUNTRY, LocationDto.REGION, LocationDto.DISTRICT);
 		}
 
+		if (FacadeProvider.getCountryFacade().getServerCountry() == null) {
+			setEnabled(false, LocationDto.REGION, LocationDto.DISTRICT, LocationDto.COMMUNITY);
+		} ;
+
+		country.addValueChangeListener(e -> {
+			CountryReferenceDto serverCountryDto = FacadeProvider.getCountryFacade().getServerCountry();
+			CountryReferenceDto countryDto = (CountryReferenceDto) e.getProperty().getValue();
+			if (countryDto != null && serverCountryDto != null && serverCountryDto.getIsoCode().equalsIgnoreCase(countryDto.getIsoCode())) {
+				setEnabled(true, LocationDto.REGION, LocationDto.DISTRICT, LocationDto.COMMUNITY);
+			} else {
+				setEnabled(false, LocationDto.REGION, LocationDto.DISTRICT, LocationDto.COMMUNITY);
+				region.setValue(null);
+				district.setValue(null);
+				community.setValue(null);
```

Comment:
Facility needs to be set to null as well

Reply:
Do we also need to disable facility?

Reply:
Yes, basically all infrastructure fields need to be disabled when a foreign country is selected